### PR TITLE
Make TYPO3 9.5 compatible

### PR DESCRIPTION
**Pull request by bjo3rnf**
Hi,

this patch addresses #113 and removes most deprecations. What I am not sure about is how to handle the deprecated `EidUtility` calls and how to properly match the backend route in `ButtonsBarHook`. The extensions seems to work like this though in 9.5. Any hints how to solve the remaining issues are highly appreciated.

Cheers
Björn

### CODE REVIEW

**Comment by danielDigitalArt** [comment]
Hi,
i tried your Pull here and i got it working in a Typo3 9.5.2. The only issues i have found so far were the missing part if the Context-Menu, so i tried to add the Folders with groups directly in the database. Later i found that there appeared a button right next to the Button to add new files in the File-List's Right Window/Frame, which allows to change Folder Permissions when you have selected a folder inside the secured storage.

Another Problem seems to be, that the Extension right now only checks if the user is logged in or not, i have multiple usergroups set up, each folder in the storage has a setting for one of these Groups, i removed all usergroups from my test-login, except the permission for being registered. When i call the generated link i can still view the file. Guess that i have either done something wrong here or the Extension doesnt respect the groups at all.

File: Classes/Controller/BePublicUrlController.php [hunk 1]
```diff
@@ -67,7 +67,7 @@ public function dumpFile(ServerRequestInterface $request, ResponseInterface $res
             }
 
             ob_start();
-            $file->getStorage()->dumpFileContents($file);
+            $file->getStorage()->streamFile($file);
```
Comment (frans-beech-it) [inline Classes/Controller/BePublicUrlController.php]: does not exist in 8LTS yet, so we need to check if the method exists?

File: Classes/Configuration/ExtensionConfiguration.php [hunk 2]
```diff
@@ -42,14 +45,13 @@ private static function init()
...
-
-            $extensionConfig = unserialize($GLOBALS['TYPO3_CONF_VARS']['EXT']['extConf']['fal_securedownload']);
+            $extensionConfig = GeneralUtility::makeInstance(ExtensionConfigurationCore::class)->get('fal_securedownload');
             self::$loginRedirectUrl = $extensionConfig['login_redirect_url'];
             self::$noAccessRedirectUrl = $extensionConfig['no_access_redirect_url'];
             self::$forceDownload = (bool)$extensionConfig['force_download'];
             self::$forceDownloadForExt = $extensionConfig['force_download_for_ext'];
             self::$trackDownloads = (bool)$extensionConfig['track_downloads'];
-            self::$resumableDownload = (bool)(isset($extensionConfig['resumable_download']) ? $extensionConfig['resumable_download'] : false);
+            self::$resumableDownload = (isset($extensionConfig['resumable_download']) ? $extensionConfig['resumable_download'] : false);
```
Comment (frans-beech-it) [inline Classes/Configuration/ExtensionConfiguration.php]: Why did you drop the `(bool)`?

File: Classes/Configuration/ExtensionConfiguration.php [hunk 3]
```diff
@@ -42,14 +45,13 @@ private static function init()
     {
         if (!self::$isInitialized) {
             self::$isInitialized = true;
-
-            $extensionConfig = unserialize($GLOBALS['TYPO3_CONF_VARS']['EXT']['extConf']['fal_securedownload']);
+            $extensionConfig = GeneralUtility::makeInstance(ExtensionConfigurationCore::class)->get('fal_securedownload');
```
Comment (frans-beech-it) [inline Classes/Configuration/ExtensionConfiguration.php]: this class doesn't exist yet in 8LTS

File: Classes/Hooks/FileDumpHook.php [hunk 4]
```diff
@@ -208,7 +201,7 @@ protected function dumpFileContents($file, $asDownload, $resumableDownload)
         }
 
         if (!$resumableDownload) {
-            $file->getStorage()->dumpFileContents($file, $asDownload, $downloadName);
+            $file->getStorage()->streamFile($file, $asDownload, $downloadName);
```
Comment (frans-beech-it) [inline Classes/Hooks/FileDumpHook.php]: does not exist in 8LTS yet, so we need to check if the method exists?

File: Classes/Hooks/DocHeaderButtonsHook.php [hunk 5]
```diff
@@ -84,7 +84,7 @@ public function getButtons($params, ButtonBar $buttonBar)
     {
         $buttons = $params['buttons'];
 
-        if (GeneralUtility::_GP('M') === 'file_FilelistList') {
+        if (GeneralUtility::_GP('route') === '/file/FilelistList/') {
```
Comment (frans-beech-it) [inline Classes/Hooks/DocHeaderButtonsHook.php]: this is not available yet in 8LTS

File: Classes/Configuration/ExtensionConfiguration.php (hunk 2 above)
Comment (frans-beech-it) [reply Classes/Configuration/ExtensionConfiguration.php]: moved it a little

**Comment by frans-beech-it** [comment]
> Another Problem seems to be, that the Extension right now only checks if the user is logged in or not, i have multiple usergroups set up, each folder in the storage has a setting for one of these Groups, i removed all usergroups from my test-login, except the permission for being registered. When i call the generated link i can still view the file. Guess that i have either done something wrong here or the Extension doesnt respect the groups at all.

That works as expected. Maybe a configuration error on your side?

I added a group restriction to a folder. When the user is in that group he sees the files when he isn't he doesn't

**Comment by frans-beech-it** [comment]
the depredations can only be resolved once we drop 8LTS support or by adding a lot of extra/double code. So for now we will have live with them :smiley:

**Comment by frans-beech-it** [comment]
@bjo3rnf tnx for the work :+1:

**Comment by bjo3rnf** [comment]
Hey Frans,

thanks for taking the time, much appreciated.

Doei!

**Pull request merged**